assets/src/editor/events/dates-and-tickets-metabox/dates-and-tickets-metabox.js

```diff
@@ -20,108 +20,101 @@ import {
+					>
+						<PanelRow
+							className="ee-editor-event-dates ee-form-row">
+							<div>
+								<EditorDateEntitiesList
+									loading={ loadingDateEntities }
+									entities={ filteredDateEntities }
+									allDateEntities={ allDateEntities }
+									allTicketEntities={ allTicketEntities }
+									prefiltered
+									for="event-dates-metabox"
+									{ ...otherProps }
+								/>
+							</div>
+						</PanelRow>
+					</PanelBody>
+				</Panel>
+				<br />
+				<h1 className="ee-metabox-heading">
+					<Dashicon icon="tickets-alt" />
+					{ __( 'Available Tickets', 'event_espresso' ) }
+				</h1>
+				<Panel>
+					<PanelBody
+						id={ `ee-editor-event-tickets-${ eventId }` }
+						className="ee-editor-event-tickets espresso-editor"
 					>
-						<h1 className="ee-metabox-heading">
-							<EspressoIcon icon="calendar" />
-							{ __( 'Event Dates', 'event_espresso' ) }
-						</h1>
-						<Panel>
-							<PanelBody
-								id={ `ee-editor-event-dates-${ eventId }` }
-								className="ee-editor-event-dates espresso-editor"
-							>
-								<PanelRow
-									className="ee-editor-event-dates ee-form-row">
-									<div>
-										<EditorDatesList
-											loading={ loadingDates }
-											entities={ datetimes }
-											allDates={ allDates }
-											allTickets={ allTickets }
-											prefiltered
-											for="event-dates-metabox"
-											{ ...otherProps }
-										/>
-									</div>
-								</PanelRow>
-							</PanelBody>
-						</Panel>
-						<br />
-						<h1 className="ee-metabox-heading">
-							<Dashicon icon="tickets-alt" />
-							{ __( 'Available Tickets', 'event_espresso' ) }
-						</h1>
-						<Panel>
-							<PanelBody
-								id={ `ee-editor-event-tickets-${ eventId }` }
-								className="ee-editor-event-tickets espresso-editor"
-							>
-								<PanelRow
-									className="ee-editor-event-tickets ee-form-row">
-									<div>
-										<EditorTicketsList
-											loading={ loadingTickets }
-											entities={ tickets }
-											allDates={ allDates }
-											isChained={ isChained }
-											prefiltered
-											for="event-tickets-metabox"
-											{ ...otherProps }
-										/>
-									</div>
-								</PanelRow>
-							</PanelBody>
-						</Panel>
-						<br />
-					</FormContainer>
-				</Fragment>
-			);
-		} }
-	/>
-);
+						<PanelRow
+							className="ee-editor-event-tickets ee-form-row">
+							<div>
+								<EditorTicketEntitiesList
+									loading={ loadingTicketEntities }
+									entities={ filteredTicketEntities }
+									allDateEntities={ allDateEntities }
+									isChained={ isChained }
+									prefiltered
+									for="event-tickets-metabox"
+									{ ...otherProps }
+								/>
+							</div>
+						</PanelRow>
+					</PanelBody>
+				</Panel>
+				<br />
+			</FormContainer>
+		</Fragment>
+	);
+};
+
+const DatesAndTicketsMetabox = withEventDateAndTicketEntityFilterState( MetaBox );
```

Comment:
I introduced the `withEventDateAndTicketEntityFilterState` hoc and used that here instead of the render prop.  The main reason is because a future refactor will likely implement hooks and this brings us one more step close to that.